azure/datalake/store/core.py

```diff
@@ -899,7 +913,10 @@ def flush(self, force=False):
                         path=self.path.as_posix(),
                         data=d2,
                         overwrite='true',
-                        write='true')
+                        write='true',
+                        syncFlag=syncFlag,
```

Comment:
You cannot just pass syncFlag as-is from the caller. If the buffer (data[] array) has more than 4MB, then this method breaks it up into 4MB chunks. In that case, only the last chunk should send syncFlag=CLOSE; all others before that should send syncFlag=DATA.